### ISSUE TITLE
Supabase dev service no longer starting, vector container keeps restarting

### ISSUE THREAD

**Issue by sosensible** [supabase/cli#2538]
**Describe the bug**
Using supabase cli to do local dev. The start fails with the vector container restarting until healthcheck fails.

**To Reproduce**
1. run > supabase start  ... in project that was working
2. starts until the vector container quits, starts, quits, starts, ... health check fail
3. failed running this also, still attempted health check > supabase start --ignore-health-check

**Expected behavior**
Supabase services to start local like they used to.

**Screenshots**
n/a

**System information**
Rerun the failing command with `--create-ticket` flag.
 - Ticket ID: e8fd446f149a4b67aab8ce16d71549fa
 - Version of OS: Windows 11
 - Version of CLI: 1.183.5
 - Version of Docker: 27.0.3
 - Versions of services: 
 ```
        SERVICE IMAGE      │      LOCAL       │ LINKED
  ─────────────────────────┼──────────────────┼─────────
    supabase/postgres      │ 15.1.1.61        │ -
    supabase/gotrue        │ v2.151.0         │ -
    postgrest/postgrest    │ v12.2.0          │ -
    supabase/realtime      │ v2.29.15         │ -
    supabase/storage-api   │ v1.0.6           │ -
    supabase/edge-runtime  │ v1.54.10         │ -
    supabase/studio        │ 20240701-05dfbec │ -
    supabase/postgres-meta │ v0.83.2          │ -
    supabase/logflare      │ 1.4.0            │ -
    supabase/supavisor     │ 1.1.56           │ -
    darthsim/imgproxy      │ v3.8.0           │ -
```
**Additional context**
n/a

**Comment by sosensible** [comment]
I use Docker for multiple things; nothing else has been an issue ATM or recently.

**Comment by hemanshum** [comment]
I'm experiencing the same issue when trying to start it on Windows; however, it works fine on Mac.

**Comment by sweatybridge** [comment]
The workaround for now is to turn off analytics in config.toml

```
[analytics]
enabled = false
```

**Comment by sweatybridge** [comment]
I've tested a fix for this in cli 1.187.5. You will need to enable a few settings on docker desktop on windows (see screenshot).

- expose daemon on tcp
- allow *.docker.internal names

![settings](https://github.com/user-attachments/assets/9aead998-401b-4bbd-bdbc-46611d8cb4e0)

After that, you can enable analytics without running into crashloop. The postgres and api gateway logs are still unavailable on studio but I will look into that separately.

**Comment by sosensible** [comment]
This should be reopened, it did not fix things. I created the setting above.
* config.toml > analytics false
* set the docker settings you outlined

The local supabase starts, and in less than a minute, after pulling current images. 

P.S. We used scoop to install supabase and are running on the current update.

**Comment by jorgeguberte** [comment]
@sweatybridge that worked for me. I also enabled the settings in Docker. Thank you!

**Comment by SebasScript** [comment]
supabase@beta is working on windows 11. Thanks for for updates!

**Comment by sosensible** [comment]
For us, supabase@beta is not working on Windows 11. Everything starts up, and with or without the ignore health check flag, it still shuts right down after everything starts up.

**Comment by jdgamble555** [comment]
I'm getting this issue again in Supabase `1.191.3`.

**Comment by 1WorldCapture** [comment]
supabase@beta works for me only when I set "analytics=false", in this way vector container will not be started.

**Comment by bradrisse** [comment]
**TLDR: only use alphanumeric for your project_id in config.toml**

I had the same issue, tried upgrading to the beta version, tried analytics false which worked, but then had other issues, tried with health checks off, but then i noticed i had an odd project_id in my configtoml. 

My project_id in my config.toml was `project_id = "project-v-2.0.0"`

I change my project_id to "project" and everything magically worked!

Some errors in docker from supabase vector were ```thread 'main' panicked at 'invalid authority: InvalidIpv4Address', src/sinks/util/uri.rs:122:69```

My guess is the project_id, if it has too many periods in the text, causes some issue with the ip address when calling the analytics service.

**Comment by mdfu** [comment]
looks like analytics will be disabled for windows users for the foreseeable future?

**Comment by sweatybridge** [comment]
@mdfu analytics can be enabled on windows with a few extra docker settings. Please refer to this comment https://github.com/supabase/cli/issues/2538#issuecomment-2246855270

**Comment by mdfu** [comment]
> [@mdfu](https://github.com/mdfu) analytics can be enabled on windows with a few extra docker settings. Please refer to this comment [#2538 (comment)](https://github.com/supabase/cli/issues/2538#issuecomment-2246855270)

Thanks. Not sure why I missed that.

**Comment by rutvikgulhane** [comment]
> **TLDR: only use alphanumeric for your project_id in config.toml**
> 
> I had the same issue, tried upgrading to the beta version, tried analytics false which worked, but then had other issues, tried with health checks off, but then i noticed i had an odd project_id in my config.toml.
> 
> My project_id in my config.toml was `project_id = "project-v-2.0.0"`
> 
> I change my project_id to "project" and everything magically worked!
> 
> Some errors in docker from supabase vector were `thread 'main' panicked at 'invalid authority: InvalidIpv4Address', src/sinks/util/uri.rs:122:69`
> 
> My guess is the project_id, if it has too many periods in the text, causes some issue with the ip address when calling the analytics service.

Surprising this worked given this was a fix from 2024

**Comment by kryptonian41** [comment]
> > **TLDR: only use alphanumeric for your project_id in config.toml**
> > I had the same issue, tried upgrading to the beta version, tried analytics false which worked, but then had other issues, tried with health checks off, but then i noticed i had an odd project_id in my config.toml.
> > My project_id in my config.toml was `project_id = "project-v-2.0.0"`
> > I change my project_id to "project" and everything magically worked!
> > Some errors in docker from supabase vector were `thread 'main' panicked at 'invalid authority: InvalidIpv4Address', src/sinks/util/uri.rs:122:69`
> > My guess is the project_id, if it has too many periods in the text, causes some issue with the ip address when calling the analytics service.
> 
> Surprising this worked given this was a fix from 2024

This solution really still works, this should be added surfaced somewhere in the docs.

**Comment by cesnietor** [comment]
thanks @bradrisse I was struggling till I found your solution, thanks for sharing.

**Comment by singh-inder** [comment]
This seems to be an issue with vector's url parsing logic. periods work but only if you don't add a number at the end. For example, if you try this `project_id="myapp-2.a"` it works. This also works `project_id = "myapp-2.0.a"`. Tested on linux.
